### PR TITLE
PAPP-21216: Support for secondary device

### PR DESCRIPTION
**Pull request by pzhou-splunk**
Closes: PAPP-21216

### CODE REVIEW

File: ciscoise_connector.py [hunk 1]
```diff
@@ -648,24 +670,34 @@ def _clear_policy(self, param):
...
+    def _test_connectivity_to_device(self, device_domain, verify=True):
         try:
+            rest_endpoint = 'https://{0}{1}'.format(device_domain, ACTIVE_LIST_REST)
+            self.save_progress(phantom.APP_PROG_CONNECTING_TO_ELLIPSES, device_domain)
             resp = requests.get(rest_endpoint, auth=self._auth, verify=verify)
         except Exception as e:
             return self.set_status(phantom.APP_ERROR, CISCOISE_ERR_TEST_CONNECTIVITY_FAILED, e)
 
-        if resp.status_code != 200:
+        if resp.status_code == 200:
```
Comment (ngaglani-crest) [inline ciscoise_connector.py:681]: What's the reason for this change.

File: ciscoise.json [hunk 2]
```diff
@@ -64,6 +64,11 @@
             "order": 5,
             "description": "Password for ERS APIs",
             "required": true
+        },
+        "secondary_device": {
```
Comment (ngaglani-crest) [inline ciscoise.json:68]: As per the ticket's description, shouldn't this be `ha-device` 🤔 ?

File: ciscoise_connector.py [hunk 3]
```diff
@@ -648,24 +670,34 @@ def _clear_policy(self, param):
...
+            return self.set_status_save_progress(phantom.APP_SUCCESS, CISCOISE_SUCC_TEST_CONNECTIVITY_PASSED)
```
Comment (ngaglani-crest) [inline ciscoise_connector.py:682]: Can we have separate success messages for both primary and secondary devices?

File: ciscoise_connector.py [hunk 4]
```diff
@@ -46,22 +46,42 @@ def __init__(self):
...
+            self._secondary_device = 'https://{0}'.format(self._secondary_device)
+            self._call_ers_api = self._secondary_device_wrapper(self._call_ers_api)
+            self._call_rest_api = self._secondary_device_wrapper(self._call_rest_api)
 
-    def _call_ers_api(self, endpoint, action_result, data=None, allow_unknown=True, method="get"):
+        return phantom.APP_SUCCESS
 
+    def _secondary_device_wrapper(self, func):
+        def make_another_call(*args, **kwargs):
+            self.debug_print("Making call to first device")
```
Comment (ngaglani-crest) [inline ciscoise_connector.py:72]: Can we change first to primary, and second to secondary?

File: ciscoise_connector.py [hunk 5]
```diff
@@ -91,9 +111,11 @@ def _call_ers_api(self, endpoint, action_result, data=None, allow_unknown=True,
 
         return phantom.APP_SUCCESS, ret_data
 
-    def _call_rest_api(self, endpoint, action_result, schema=None, data=None, allow_unknown=True):
+    def _call_rest_api(self, endpoint, action_result, schema=None, data=None, allow_unknown=True, try_secondary_device=False):
```
Comment (ngaglani-crest) [inline ciscoise_connector.py:114]: Instead of using the `try_secondary_device` flag, how about we use a required variable? Something like, base_url. When we call the method for the second time, then instead of passing try_secondary_device, we can simply pass `base_url=self. _secondary_device` and for the first call we can pass `base_url=self. _base_url`.

File: ciscoise_connector.py (hunk 3 above)
Comment (pdesai-crest) [reply ciscoise_connector.py:682]: This is a helper method for test_connectivity. This method is called from test_connectivity once if only one node is present and twice if two nodes are present. And message is displayed for each node according to return value of this method. Actual test connectivity is below at line 686

**Comment by pdesai-crest** [comment]
Closing as this is covered in #2
